### PR TITLE
Move answer result and collection result into JsonModel

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
This is intended to help with unravelling the dependencies between the mobile passive data library and storing the results in the top level task result object.

### CODE REVIEW

File: Sources/JsonModel/ResultData/CollectionResult.swift [hunk 1]
```diff
@@ -33,26 +33,73 @@
...
+    }
+    
+    /// Find a result within this collection.
+    /// - parameter identifier: The identifier associated with the result.
+    /// - returns: The result or `nil` if not found.
+    func findResult(with identifier: String) -> ResultData? {
+        return self.children.first(where: { $0.identifier == identifier })
+    }
+    
+    /// Insert the result at the end of the `children`, replacing the previous instance with the same identifier.
```
Comment (Erin-Mounts) [inline Sources/JsonModel/ResultData/CollectionResult.swift:58]:
```suggestion
    /// Insert the result at the end of the `children` collection, or replace the previous instance with the
    /// same identifier if one already exists.
```

File: Sources/JsonModel/ResultData/ErrorResult.swift [hunk 2]
```diff
@@ -2,7 +2,7 @@
 //  ErrorResultObject.swift
 //  
 //
-//  Copyright © 2020 Sage Bionetworks. All rights reserved.
+//  Copyright © 2017-2021 Sage Bionetworks. All rights reserved.
```
Comment (Erin-Mounts) [inline Sources/JsonModel/ResultData/ErrorResult.swift:5]: Is 2017 the actual original date of this file, or should it be 2020 (as in the previous copyright)?

File: Sources/JsonModel/ResultData/CollectionResult.swift [hunk 3]
```diff
@@ -2,7 +2,7 @@
 //  CollectionResultObject.swift
 //  
 //
-//  Copyright © 2020 Sage Bionetworks. All rights reserved.
+//  Copyright © 2017-2021 Sage Bionetworks. All rights reserved.
```
Comment (Erin-Mounts) [inline Sources/JsonModel/ResultData/CollectionResult.swift:5]: 2017 or 2020 (previous copyright)?

File: Sources/JsonModel/ResultData/FileResult.swift [hunk 4]
```diff
@@ -2,7 +2,7 @@
 //  FileResultObject.swift
 //  
 //
-//  Copyright © 2020 Sage Bionetworks. All rights reserved.
+//  Copyright © 2017-2021 Sage Bionetworks. All rights reserved.
```
Comment (Erin-Mounts) [inline Sources/JsonModel/ResultData/FileResult.swift:5]: 2017 or 2020 (previous copyright)?

File: Sources/JsonModel/ResultData/ResultData.swift [hunk 5]
```diff
@@ -1,7 +1,7 @@
 //
 //  ResultData.swift
 //
-//  Copyright © 2020 Sage Bionetworks. All rights reserved.
+//  Copyright © 2017-2021 Sage Bionetworks. All rights reserved.
```
Comment (Erin-Mounts) [inline Sources/JsonModel/ResultData/ResultData.swift:4]: 2017 or 2020 (previous copyright)?

File: Sources/JsonModel/ResultData/ResultDataSerializer.swift [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        if name == SerializableResultType.jsonValue.rawValue {
+            print("WARNING!!! \(name) is deprecated. Replace with `answer`.")
+            return SerializableResultType.answer.stringValue
+        }
+        else {
+            return name
+        }
+    }
+    
+    public func add(_ example: SerializableResultData) {
```
Comment (Erin-Mounts) [inline Sources/JsonModel/ResultData/ResultDataSerializer.swift:129]: doc comment?

**Review by Erin-Mounts — APPROVED** [review]
Bunch o' questions, one or two suggestions, otherwise looks good

File: Sources/JsonModel/ResultData/ErrorResult.swift (hunk 2 above)
Comment (syoung-smallwisdom) [reply Sources/JsonModel/ResultData/ErrorResult.swift:5]: This file now includes the *protocol* as well as an object that conforms to that protocol. The protocol was written in 2017.

File: Sources/JsonModel/ResultData/CollectionResult.swift (hunk 3 above)
Comment (syoung-smallwisdom) [reply Sources/JsonModel/ResultData/CollectionResult.swift:5]: This file now includes the *protocol* as well as an object that conforms to that protocol. The protocol was written in 2017.

File: Sources/JsonModel/ResultData/FileResult.swift (hunk 4 above)
Comment (syoung-smallwisdom) [reply Sources/JsonModel/ResultData/FileResult.swift:5]: This file now includes the *protocol* as well as an object that conforms to that protocol. The protocol was written in 2017.

File: Sources/JsonModel/ResultData/ResultData.swift (hunk 5 above)
Comment (syoung-smallwisdom) [reply Sources/JsonModel/ResultData/ResultData.swift:4]: This file now includes the *protocol* as well as an object that conforms to that protocol. The protocol was written in 2017.

**Pull request merged**